File: BasicsWithJava/BasicCalculator/app/src/main/java/com/sumere/basiccalculator/MainActivity.java
package com.sumere.basiccalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText num1Text;
    EditText num2Text;
    EditText resultText;

    public void calculate(String operator){
        switch(operator){
            case "+":
                break;
            case "-":
                break;
            case "*":
                break;
            case "/":
                break;
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        num1Text = findViewById(R.id.number1Text);
        num2Text = findViewById(R.id.number2Text);
        resultText = findViewById(R.id.resultText);
    }

    public void add(View view){
        double num1 = Double.parseDouble(num1Text.getText().toString());
        double num2 = Double.parseDouble(num2Text.getText().toString());
        double result = num1 + num2;
        resultText.setText("Result: " + result);
    }

    public void subtract(View view){
        double num1 = Double.parseDouble(num1Text.getText().toString());
        double num2 = Double.parseDouble(num2Text.getText().toString());
        double result = num1 - num2;
        resultText.setText("Result: " + result);
    }

    public void multiply(View view){
        double num1 = Double.parseDouble(num1Text.getText().toString());
        double num2 = Double.parseDouble(num2Text.getText().toString());
        double result = num1 * num2;
        resultText.setText("Result: " + result);
    }

    public void divide(View view){
        double num1 = Double.parseDouble(num1Text.getText().toString());
        double num2 = Double.parseDouble(num2Text.getText().toString());
        double result;
        if(num2 != 0)
            result = num1 / num2;
        else
            result = 0;
        resultText.setText("Result: " + result);
    }
}

File: BasicCalculatorKotlin/app/src/main/java/com/sumere/basiccalculatorkotlin/MainActivity.kt
package com.sumere.basiccalculatorkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.sumere.basiccalculatorkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding : ActivityMainBinding
    private var number1 : Double? = null
    private var number2 : Double? = null
    private var result : Any = 0.0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(this.layoutInflater)
        //val view = binding.root
        setContentView(binding.root)

    }

    private fun getNumbers(){
        number2 = binding.num2TextView.text.toString().toDoubleOrNull()
        number1 = binding.num1TextView.text.toString().toDoubleOrNull()
    }

    fun addNumbers(view : View){
        getNumbers()
        if(number1 != null && number2 != null) {
            result = number1!! + number2!!
        } else{
            result = "Invalid Input"
        }
        binding.resultLabel.text = "Result: ${result}"
    }

    fun subtractNumbers(view : View){
        getNumbers()
        if(number1 != null && number2 != null) {
            result = number1!! - number2!!
        } else{
            result = "Invalid Input"
        }
        binding.resultLabel.text = "Result: ${result}"
    }

    fun multiplyNumbers(view: View){
        getNumbers()
        if(number1 != null && number2 != null) {
            result = number1!! * number2!!
        } else{
            result = "Invalid Input"
        }
        binding.resultLabel.text = "Result: ${result}"
    }
    fun divideNumbers(view : View){
        getNumbers()
        if(number1 != null && number2 != null) {
            result = number1!! / number2!!
        } else{
            result = "Invalid Input"
        }
        binding.resultLabel.text = "Result: ${result}"
    }
}